src/api.rs

```diff
@@ -0,0 +1,119 @@
+use std::sync::Arc;
+
+use anyhow::{anyhow, Result};
+use compositor_common::{
+    scene::{InputId, OutputId, Resolution, SceneSpec},
+    transformation::{TransformationRegistryKey, TransformationSpec},
+};
+use compositor_pipeline::pipeline;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    rtp_receiver::{self, RtpReceiver},
+    rtp_sender::{self, EncoderSettings, RtpSender},
+};
+
+pub type Pipeline = compositor_pipeline::Pipeline<RtpReceiver, RtpSender>;
+
+#[derive(Serialize, Deserialize)]
+pub struct RegisterInputRequest {
+    pub id: InputId,
+    pub port: u16,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct RegisterOutputRequest {
+    pub id: OutputId,
+    pub port: u16,
+    pub ip: String,
+    pub resolution: Resolution,
+    pub encoder_settings: EncoderSettings,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum Request {
+    RegisterInput(RegisterInputRequest),
+    RegisterOutput(RegisterOutputRequest),
+    UnregisterInput {
+        id: InputId,
+    },
+    UnregisterOutput {
+        id: OutputId,
+    },
+    UpdateScene(Arc<SceneSpec>),
+    RegisterTransformation {
+        key: TransformationRegistryKey,
+        transform: TransformationSpec,
+    },
+    Init(pipeline::Options),
+    Start,
+}
```

Comment:
[NIT] It doesn't matter at all, but `api.rs` file seem like the first one I would checkout out in such a project, so this order makes more sense (especially since out HTTP server is stateful)
```suggestion
#[derive(Serialize, Deserialize)]
#[serde(tag = "type", rename_all = "snake_case")]
pub enum Request {
    Init(pipeline::Options),
    RegisterInput(RegisterInputRequest),
    UnregisterInput {
        id: InputId,
    },
    RegisterOutput(RegisterOutputRequest),
    UnregisterOutput {
        id: OutputId,
    },
    RegisterTransformation {
        key: TransformationRegistryKey,
        transform: TransformationSpec,
    },
    UpdateScene(Arc<SceneSpec>),
    Start,
}
```